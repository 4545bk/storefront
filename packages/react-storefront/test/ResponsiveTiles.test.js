/**
 * @license
 * Copyright Â© 2017-2018 Moov Corporation.  All rights reserved.
 */
import React from 'react'
import ResponsiveTiles from '../src/ResponsiveTiles'
import { mount } from 'enzyme'

describe('ResponsiveTiles', () => {
  it('should render', () => {
    expect(mount(
      <ResponsiveTiles>
        <div>Tile 1</div>
        <div>Tile 2</div>
        <div>Tile 3</div>
        <div>Tile 4</div>
      </ResponsiveTiles>
    )).toMatchSnapshot()
  })
  it('should skip invalid elements', () => {
    expect(mount(
      <ResponsiveTiles>
        <div>Tile 1</div>
        Foo
        <div>Tile 3</div>
        <div>Tile 4</div>
      </ResponsiveTiles>
    )).toMatchSnapshot()
  })
})