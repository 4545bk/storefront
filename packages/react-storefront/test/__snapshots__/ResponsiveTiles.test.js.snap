// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResponsiveTiles should render 1`] = `
<ResponsiveTiles cols={{...}} spacing={15}>
  <WithStyles(ResponsiveTilesInner)>
    <ResponsiveTilesInner classes={{...}}>
      <ul className="RSFResponsiveTiles-root-1">
        <WithStyles(GridListTile) classes={{...}}>
          <GridListTile classes={{...}} cols={1} component="li" rows={1}>
            <li className="MuiGridListTile-root-3 RSFResponsiveTiles-tile-2">
              <EventListener target="window" onResize={[Function: debounced]} />
              <div className="MuiGridListTile-tile-4">
                <div>
                  Tile 1
                </div>
              </div>
            </li>
          </GridListTile>
        </WithStyles(GridListTile)>
        <WithStyles(GridListTile) classes={{...}}>
          <GridListTile classes={{...}} cols={1} component="li" rows={1}>
            <li className="MuiGridListTile-root-3 RSFResponsiveTiles-tile-2">
              <EventListener target="window" onResize={[Function: debounced]} />
              <div className="MuiGridListTile-tile-4">
                <div>
                  Tile 2
                </div>
              </div>
            </li>
          </GridListTile>
        </WithStyles(GridListTile)>
        <WithStyles(GridListTile) classes={{...}}>
          <GridListTile classes={{...}} cols={1} component="li" rows={1}>
            <li className="MuiGridListTile-root-3 RSFResponsiveTiles-tile-2">
              <EventListener target="window" onResize={[Function: debounced]} />
              <div className="MuiGridListTile-tile-4">
                <div>
                  Tile 3
                </div>
              </div>
            </li>
          </GridListTile>
        </WithStyles(GridListTile)>
        <WithStyles(GridListTile) classes={{...}}>
          <GridListTile classes={{...}} cols={1} component="li" rows={1}>
            <li className="MuiGridListTile-root-3 RSFResponsiveTiles-tile-2">
              <EventListener target="window" onResize={[Function: debounced]} />
              <div className="MuiGridListTile-tile-4">
                <div>
                  Tile 4
                </div>
              </div>
            </li>
          </GridListTile>
        </WithStyles(GridListTile)>
      </ul>
    </ResponsiveTilesInner>
  </WithStyles(ResponsiveTilesInner)>
</ResponsiveTiles>
`;

exports[`ResponsiveTiles should skip invalid elements 1`] = `
<ResponsiveTiles cols={{...}} spacing={15}>
  <WithStyles(ResponsiveTilesInner)>
    <ResponsiveTilesInner classes={{...}}>
      <ul className="RSFResponsiveTiles-root-7">
        <WithStyles(GridListTile) classes={{...}}>
          <GridListTile classes={{...}} cols={1} component="li" rows={1}>
            <li className="MuiGridListTile-root-3 RSFResponsiveTiles-tile-8">
              <EventListener target="window" onResize={[Function: debounced]} />
              <div className="MuiGridListTile-tile-4">
                <div>
                  Tile 1
                </div>
              </div>
            </li>
          </GridListTile>
        </WithStyles(GridListTile)>
        <WithStyles(GridListTile) classes={{...}}>
          <GridListTile classes={{...}} cols={1} component="li" rows={1}>
            <li className="MuiGridListTile-root-3 RSFResponsiveTiles-tile-8">
              <EventListener target="window" onResize={[Function: debounced]} />
              <div className="MuiGridListTile-tile-4">
                <div>
                  Tile 3
                </div>
              </div>
            </li>
          </GridListTile>
        </WithStyles(GridListTile)>
        <WithStyles(GridListTile) classes={{...}}>
          <GridListTile classes={{...}} cols={1} component="li" rows={1}>
            <li className="MuiGridListTile-root-3 RSFResponsiveTiles-tile-8">
              <EventListener target="window" onResize={[Function: debounced]} />
              <div className="MuiGridListTile-tile-4">
                <div>
                  Tile 4
                </div>
              </div>
            </li>
          </GridListTile>
        </WithStyles(GridListTile)>
      </ul>
    </ResponsiveTilesInner>
  </WithStyles(ResponsiveTilesInner)>
</ResponsiveTiles>
`;
